x-pack/platform/plugins/shared/ml/public/application/explorer/annotations_state_service.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Observable } from 'rxjs';
+import {
+  BehaviorSubject,
+  combineLatest,
+  startWith,
+  map,
+  switchMap,
+  Subscription,
+  lastValueFrom,
+} from 'rxjs';
+import type { TimefilterContract } from '@kbn/data-plugin/public';
+import type { TimeRangeBounds } from '@kbn/ml-time-buckets';
+import { mlTimefilterRefresh$ } from '@kbn/ml-date-picker';
+import { extractErrorMessage } from '@kbn/ml-error-utils';
+import type { AnnotationsTable } from '../../../common/types/annotations';
+import type { Annotations } from '../../../common/types/annotations';
+import type { AnomalyExplorerCommonStateService } from './anomaly_explorer_common_state';
+import type { AnomalyTimelineStateService } from './anomaly_timeline_state_service';
+import type { MlApi } from '../services/ml_api_service';
+import { StateService } from '../services/state_service';
+import { ANNOTATIONS_TABLE_DEFAULT_QUERY_SIZE } from '../../../common/constants/search';
+import { getSelectionTimeRange, type ExplorerJob } from './explorer_utils';
+import { loadAnnotationsTableData } from './explorer_utils';
+import type { Refresh } from '../routing/use_refresh';
+
+/**
+ * Dedicated state service for annotations.
+ * - overallAnnotations$ drives the swimlane overlay
+ * - annotationsTable$ drives the Annotations panel
+ */
+export class AnnotationsStateService extends StateService {
+  private readonly _overallAnnotations$ = new BehaviorSubject<AnnotationsTable>({
+    annotationsData: [],
+    error: undefined,
+  });
+
+  private readonly _annotationsTable$ = new BehaviorSubject<AnnotationsTable>({
+    annotationsData: [],
+    totalCount: 0,
+    error: undefined,
+  });
+
+  private readonly _timeBounds$: Observable<TimeRangeBounds>;
+  private readonly _refreshSubject$: Observable<Refresh>;
+
+  constructor(
+    private readonly mlApi: MlApi,
+    private readonly timefilter: TimefilterContract,
+    private readonly anomalyExplorerCommonStateService: AnomalyExplorerCommonStateService,
+    private readonly anomalyTimelineStateService: AnomalyTimelineStateService
+  ) {
+    super();
+
+    this._timeBounds$ = this.timefilter.getTimeUpdate$().pipe(
+      startWith(null),
+      map(() => this.timefilter.getBounds())
+    );
```

Comment:
Absolute nit pick, but it looks like most classes derived from `StateService` have this code in their constructor. I'm wondering if it's work making an intermediate class do this in it's constructor, e.g. `StateServiceWithTimeBounds` (horrible name) 
It's possibly not worth doing, but it was something I noticed, so thought I'd write a comment.